pkg/apis/v1beta1/nodepool.go

```diff
@@ -190,6 +190,12 @@ type NodePool struct {
 	Status NodePoolStatus `json:"status,omitempty"`
 }
 
+// We need to bump the NodePoolHashVersion under these condition when we make an update to the NodePool CRD
```

Comment:
```suggestion
// We need to bump the NodePoolHashVersion when we make an update to the NodePool CRD under these conditions:
```

nit: consider re-ordering this wording